packages/core/src/DdSdkReactNative.tsx

```diff
@@ -38,7 +47,25 @@ export class DdSdkReactNative {
             configuration.additionalConfig[DdSdkReactNative.DD_SOURCE_KEY] = 'react-native';
             configuration.additionalConfig[DdSdkReactNative.DD_NATIVE_VIEW_TRACKING_KEY] = configuration.nativeViewTracking;
             if (configuration.verbosity != undefined) {
-                configuration.additionalConfig[DdSdkReactNative.DD_SDK_VERBOSITY_KEY] = configuration.verbosity
+                configuration.additionalConfig[DdSdkReactNative.DD_SDK_VERBOSITY_KEY] = configuration.verbosity;
+            }
+
+            if (configuration.proxyConfig) {
+                const additionalConfig = configuration.additionalConfig;
+                const proxyConfig = configuration.proxyConfig;
+
+                additionalConfig[DdSdkReactNative.DD_PROXY_TYPE_KEY] = proxyConfig.type;
+                additionalConfig[DdSdkReactNative.DD_PROXY_ADDRESS_KEY] = proxyConfig.address;
+                additionalConfig[DdSdkReactNative.DD_PROXY_PORT_KEY] = proxyConfig.port;
+                if (proxyConfig.username && proxyConfig.password) {
+                    if (proxyConfig.type == ProxyType.SOCKS) {
+                        console.warn("SOCKS proxy configuration doesn't support Basic authentication out"
+                            + " of the box, please set it up using java.net.Authenticator by yourself.")
+                    } else {
+                        additionalConfig[DdSdkReactNative.DD_PROXY_USERNAME_KEY] = proxyConfig.username;
+                        additionalConfig[DdSdkReactNative.DD_PROXY_PASSWORD_KEY] = proxyConfig.password;
+                    }
```

Comment:
Can we still set the credential values?
```suggestion
                   additionalConfig[DdSdkReactNative.DD_PROXY_USERNAME_KEY] = proxyConfig.username;
                    additionalConfig[DdSdkReactNative.DD_PROXY_PASSWORD_KEY] = proxyConfig.password;
```

Reply:
Is it for iOS? Because on Android user has to create java-module, so credentials should be set there, not in the hybrid code.

Reply:
So on iOS you have the ability to set your proxy credentials as part of the configuration, BUT the networking layer is ignoring it. Even for Basic HTTP auth, we need to manually set [`Proxy-Authorization`](https://github.com/DataDog/dd-sdk-ios/pull/596). This issue might be fixed in futur version of the OS, or we might get an API to set the credentials for socks. So it make sense to still set the credentials here for iOS. Would it have some drawbacks on Android?